mypy/checker.py

```diff
@@ -148,6 +152,12 @@ def visit_file(self, file_node: MypyFile, path: str) -> None:
         self.weak_opts = file_node.weak_opts
         self.enter_partial_types()
         self.is_typeshed_stub = self.errors.is_typeshed_file(path)
+        if self.options.strict_optional_whitelist is None:
+            self.suppress_none_errors = False
+        else:
+            self.suppress_none_errors = not any(fnmatch.fnmatch(path, pattern)
```

Comment:
You realize that `fnmatch('a/b.py', 'a*.py')` is true, right? This can be seen as a poor-man's `**` matching, but it might also lead to surprise.